credential/manifest/model.go

```diff
@@ -209,130 +212,151 @@ func IsValidCredentialApplicationForManifest(cm CredentialManifest, applicationA
 		}
 
 		for _, format := range ca.Format.FormatValues() {
-			if _, ok := cmFormats[format]; !ok {
-				return errresp.NewErrorResponse(errresp.ApplicationError, "credential application's format must be a subset of the format property in the credential manifest")
+			if _, ok = cmFormats[format]; !ok {
+				err = errresp.NewErrorResponse(errresp.ApplicationError, "credential application's format must be a subset of the format property in the credential manifest")
+				return
 			}
 		}
 	}
 
 	if (cm.PresentationDefinition != nil && len(cm.PresentationDefinition.InputDescriptors) > 0) &&
 		(ca.PresentationSubmission == nil || len(ca.PresentationSubmission.DescriptorMap) == 0) {
-		return errresp.NewErrorResponsef(errresp.ApplicationError, "no descriptors provided for application: %s against manifest: %s", ca.ID, cm.ID)
+		err = errresp.NewErrorResponsef(errresp.ApplicationError, "no descriptors provided for application: %s against manifest: %s", ca.ID, cm.ID)
+		return
 	}
 
-	// The Credential Application object MUST contain a presentation_submission property IF the related Credential Manifest contains a presentation_definition.
-	// Its value MUST be a valid Presentation Submission:
-	if !cm.PresentationDefinition.IsEmpty() {
-		if ca.PresentationSubmission.IsEmpty() {
-			return errresp.NewErrorResponse(errresp.ApplicationError, "credential application's presentation submission cannot be empty because the credential manifest's presentation definition is not empty")
-		}
+	// The Credential Application object MUST contain a presentation_submission property IF the related Credential
+	// Manifest contains a presentation_definition. Its value MUST be a valid Presentation Submission:
+	if cm.PresentationDefinition.IsEmpty() {
+		return
```

Comment:
good call, 

I wonder if there is an edge case where if there is not a PresDef but the presentation submission exists (for some reason?) and then it is invalid

Reply:
great point - will add a check